src/platform_impl/macos/app_delegate.rs

```diff
@@ -36,10 +48,26 @@ lazy_static! {
             activation_hack::mouse_moved as extern "C" fn(&Object, Sel, id),
         );
 
+        decl.add_method(
+            sel!(application:openFile:),
+            application_open_file as extern "C" fn(&Object, Sel, id, id) -> BOOL,
+        );
+        decl.add_method(
+            sel!(application:openFiles:),
+            application_open_files as extern "C" fn(&Object, Sel, id, id),
+        );
+
         AppDelegateClass(decl.register())
     };
 }
 
+/// Copies the contents of the ns string into a `String` which gets returned.
+unsafe fn ns_string_to_rust(ns_string: id) -> String {
```

Comment:
Maybe move this to `macos/util/mod.rs`, next to `ns_string_id_ref` - they're kinda polar opposites, and both useful to have in general!